specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/Accounts.json

```diff
@@ -0,0 +1,910 @@
+            "name": "MetricAggregationType",
+            "values": [
+              {
+                "value": "Average"
+              },
+              {
+                "value": "Count"
+              },
+              {
+                "value": "Total"
+              }
+            ],
+            "modelAsExtensible": true
+          },
+          "description": "The metric aggregation type"
+        },
+        "dimensions": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MetricDimension"
+          },
+          "description": "The metric dimensions."
+        }
+      },
+      "type": "object",
+      "required": [
+        "unit",
+        "aggregationType"
+      ],
+      "description": "A metric emitted by service."
+    },
+    "ServiceSpecification": {
+      "properties": {
+        "metricSpecifications": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Metric"
+          },
+          "description": "List of metric specifications."
+        }
+      },
+      "type": "object",
+      "description": "The service metric specifications."
+    },
+    "MetricProperties": {
+      "properties": {
+        "serviceSpecification": {
+          "$ref": "#/definitions/ServiceSpecification",
+          "description": "The service specifications."
+        }
+      },
+      "type": "object",
+      "description": "Metric properties."
+    },
+    "Operation": {
+      "description": "An operation.",	
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The operation name."
+        },
+        "display": {
+          "$ref": "#/definitions/OperationDisplay",
+          "description": "The operation display name."
+        },
+        "origin": {
+          "type": "string",
+          "description": "Origin of the operation."
+        },
+        "properties": {
+          "$ref": "#/definitions/MetricProperties",
+          "description": "Operation properties format."
+        }
+      },
+      "type": "object",
+      "required": [
+        "name"
+      ]
+    },
+    "Location": {
+      "properties": {
+        "name": {
+          "type": "string"
+        }
+      },
+      "type": "object",
+      "required": [
+        "name"
+      ]
+    },
+    "EntityNameAvailabilityCheckOutput": {
+      "properties": {
+        "nameAvailable": {
+          "type": "boolean",
+          "description": "Specifies if the name is available."
+        },
+        "reason": {
+          "type": "string",
+          "description": "Specifies the reason if the name is not available."
+        },
+        "message": {
+          "type": "string",
+          "description": "Specifies the detailed reason if the name is not available."
+        }
+      },
+      "type": "object",
+      "required": [
+        "nameAvailable"
+      ],
+      "description": "The response from the check name availability request."
```

Comment:
`EntityNameAvailabilityCheckOutput` represents output of `Locations_CheckNameAvailability` operation, seems all properties can be readonly in this model. right?